### PR TITLE
Close selector

### PR DESCRIPTION
**Pull request by alexandrosfilios**

### CODE REVIEW

**Review by adecaro — APPROVED** [review]
LGTM

File: token/services/db/sql/tokenlock.go [hunk 1]
```diff
@@ -17,8 +17,52 @@ import (
...
+	}
+
+	baseTokenLockDB := newTokenLockDB(
+		db,
+		tokenLockTables{
+			TokenLocks: tables.TokenLocks,
+			Requests:   tables.Requests,
+		},
+	)
+	var tokenLockDB driver.TokenLockDB
```
Comment (alexandrosfilios) [inline token/services/db/sql/tokenlock.go:37]: Why did we need to use two variables?

File: token/services/selector/sherdlock/manager.go [hunk 2]
```diff
@@ -24,29 +34,62 @@ type tokenSelectorUnlocker interface {
...
+
+func (m *manager) Close(id transaction.ID) error {
 	if c, ok := m.selectorCache.Delete(id); ok {
-		return c.UnlockAll()
+		return c.Close()
+	}
+	return errors.New("selector for " + id + " not found")
+}
+
+func (m *manager) cleaner() {
```
Comment (alexandrosfilios) [inline token/services/selector/sherdlock/manager.go:85]: This we could do also with a DB job (for postgres). This way we don't have to run the cleanup code for each replica.

File: token/services/network/orion/txstatus.go [hunk 3]
```diff
@@ -101,7 +101,9 @@ func (r *RequestTxStatusResponderView) Call(context view.Context) (interface{},
 
 func (r *RequestTxStatusResponderView) process(context view.Context, request *TxStatusRequest) (*TxStatusResponse, error) {
 	if status, ok := r.statusCache.Get(request.TxID); ok && status.Status != driver.Busy {
-		return status, nil
+		if status.Status != driver.Valid || len(status.TokenRequestReference) != 0 {
```
Comment (alexandrosfilios) [inline token/services/network/orion/txstatus.go:104]: We can append this condition in the previous line I think

File: token/services/identity/msp/idemix/lm.go [hunk 4]
```diff
@@ -361,16 +363,23 @@ func (l *LocalMembership) loadFromStorage() error {
 	if err != nil {
 		return errors.WithMessage(err, "failed to get registered identities from kvs")
 	}
-	defer it.Close()
+	// copy the iterator
+	items := make([]driver3.IdentityConfiguration, 0)
```
Comment (alexandrosfilios) [inline token/services/identity/msp/idemix/lm.go:367]: We can have a TODO to use the copyiterator

**Comment by alexandrosfilios** [comment]
Taken care of as part of https://github.com/hyperledger-labs/fabric-token-sdk/pull/727.

**Pull request closed without merging**